### PR TITLE
Add relocations profile to Quarkus main build

### PR DESCRIPTION
**Pull request by pjgg**
Due to Quarkus main re-structure, some modules have been moved under the "relocations" profile.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,7 +113,7 @@ jobs:
         run: tar -xzf maven-repo.tgz -C ~
       - name: Build Quarkus CLI
         run: |
-          git clone https://github.com/quarkusio/quarkus.git && cd quarkus/devtools/cli && mvn -B -s ../../../.github/mvn-settings.xml clean install -Dquickly
+          git clone https://github.com/quarkusio/quarkus.git && cd quarkus/devtools/cli && mvn -B -s ../../../.github/mvn-settings.xml clean install -Dquickly -Prelocations
```
Comment (jsmrcka) [inline .github/workflows/ci.yml:116]: The same modification is missing in native build (line 173).

**Review by jsmrcka — CHANGES_REQUESTED** [review]

**Comment by jsmrcka** [comment]
Failing checks are unrelated:
- `http/reactive-routes` - fixed in https://github.com/quarkus-qe/quarkus-test-suite/pull/418.
- `cache/caffeine` - TODO

**Review by jsmrcka — APPROVED** [review]

**Comment by fedinskiy** [comment]
It looks like we need to apply the same changes in the framework: https://github.com/quarkus-qe/quarkus-test-framework/runs/4517286532?check_suite_focus=true

**Comment by jsmrcka** [comment]
> It looks like we need to apply the same changes in the framework: https://github.com/quarkus-qe/quarkus-test-framework/runs/4517286532?check_suite_focus=true

@fedinskiy See https://github.com/quarkus-qe/quarkus-test-framework/pull/355.

**Pull request merged**